# Thomas Hoffmann, EMBL Heidelberg, structures-it@embl.de
# use gencodes according to CUDA_CMAKE_ARCHITECUTES
#
diff -ru tsne-cuda-3.0.1/CMakeLists.txt tsne-cuda-3.0.1_cudacc/CMakeLists.txt
--- tsne-cuda-3.0.1/CMakeLists.txt	2021-12-10 02:27:07.000000000 +0100
+++ tsne-cuda-3.0.1_cudacc/CMakeLists.txt	2023-11-07 18:53:05.060718265 +0100
@@ -58,54 +58,59 @@
     set(CMAKE_CUDA_STANDARD_REQUIRED ON)
 endif()
 
-if(CUDAToolkit_VERSION_MAJOR EQUAL "10")
-    set(CUDA_ARCH
-        -gencode=arch=compute_30,code=sm_30
-        -gencode=arch=compute_35,code=sm_35
-        -gencode=arch=compute_37,code=sm_37
-        -gencode=arch=compute_50,code=sm_50
-        -gencode=arch=compute_52,code=sm_52
-        -gencode=arch=compute_60,code=sm_60
-        -gencode=arch=compute_61,code=sm_61
-        -gencode=arch=compute_70,code=sm_70
-        -gencode=arch=compute_75,code=sm_75
-    )
-elseif(CUDAToolkit_VERSION_MAJOR EQUAL "11" AND CUDAToolkit_VERSION_MINOR LESS "1")
-    set(CUDA_ARCH
-        -gencode=arch=compute_35,code=sm_35
-        -gencode=arch=compute_37,code=sm_37
-        -gencode=arch=compute_50,code=sm_50
-        -gencode=arch=compute_52,code=sm_52
-        -gencode=arch=compute_60,code=sm_60
-        -gencode=arch=compute_61,code=sm_61
-        -gencode=arch=compute_70,code=sm_70
-        -gencode=arch=compute_75,code=sm_75
-        -gencode=arch=compute_80,code=sm_80
-    )
-elseif(CUDAToolkit_VERSION_MAJOR EQUAL "11")
-    set(CUDA_ARCH
-        -gencode=arch=compute_35,code=sm_35
-        -gencode=arch=compute_37,code=sm_37
-        -gencode=arch=compute_50,code=sm_50
-        -gencode=arch=compute_52,code=sm_52
-        -gencode=arch=compute_60,code=sm_60
-        -gencode=arch=compute_61,code=sm_61
-        -gencode=arch=compute_70,code=sm_70
-        -gencode=arch=compute_75,code=sm_75
-        -gencode=arch=compute_80,code=sm_80
-        -gencode=arch=compute_86,code=sm_86
-    )
-else()
-    set(CUDA_ARCH
-        -gencode=arch=compute_30,code=sm_30
-        -gencode=arch=compute_35,code=sm_35
-        -gencode=arch=compute_37,code=sm_37
-        -gencode=arch=compute_50,code=sm_50
-        -gencode=arch=compute_52,code=sm_52
-        -gencode=arch=compute_60,code=sm_60
-        -gencode=arch=compute_61,code=sm_61
-        )
-endif()
+set(CUDA_ARCH "")
+foreach(sm IN ${CUDA_CMAKE_ARCHITECUTES})
+    set(CUDA_ARCH "-gencode=arch=compute_${sm},code=sm_${sm}")
+endforeach()
+
+#if(CUDAToolkit_VERSION_MAJOR EQUAL "10")
+#    set(CUDA_ARCH
+#        -gencode=arch=compute_30,code=sm_30
+#        -gencode=arch=compute_35,code=sm_35
+#        -gencode=arch=compute_37,code=sm_37
+#        -gencode=arch=compute_50,code=sm_50
+#        -gencode=arch=compute_52,code=sm_52
+#        -gencode=arch=compute_60,code=sm_60
+#        -gencode=arch=compute_61,code=sm_61
+#        -gencode=arch=compute_70,code=sm_70
+#        -gencode=arch=compute_75,code=sm_75
+#    )
+#elseif(CUDAToolkit_VERSION_MAJOR EQUAL "11" AND CUDAToolkit_VERSION_MINOR LESS "1")
+#    set(CUDA_ARCH
+#        -gencode=arch=compute_35,code=sm_35
+#        -gencode=arch=compute_37,code=sm_37
+#        -gencode=arch=compute_50,code=sm_50
+#        -gencode=arch=compute_52,code=sm_52
+#        -gencode=arch=compute_60,code=sm_60
+#        -gencode=arch=compute_61,code=sm_61
+#        -gencode=arch=compute_70,code=sm_70
+#        -gencode=arch=compute_75,code=sm_75
+#        -gencode=arch=compute_80,code=sm_80
+#    )
+#elseif(CUDAToolkit_VERSION_MAJOR EQUAL "11")
+#    set(CUDA_ARCH
+#        -gencode=arch=compute_35,code=sm_35
+#        -gencode=arch=compute_37,code=sm_37
+#        -gencode=arch=compute_50,code=sm_50
+#        -gencode=arch=compute_52,code=sm_52
+#        -gencode=arch=compute_60,code=sm_60
+#        -gencode=arch=compute_61,code=sm_61
+#        -gencode=arch=compute_70,code=sm_70
+#        -gencode=arch=compute_75,code=sm_75
+#        -gencode=arch=compute_80,code=sm_80
+#        -gencode=arch=compute_86,code=sm_86
+#    )
+#else()
+#    set(CUDA_ARCH
+#        -gencode=arch=compute_30,code=sm_30
+#        -gencode=arch=compute_35,code=sm_35
+#        -gencode=arch=compute_37,code=sm_37
+#        -gencode=arch=compute_50,code=sm_50
+#        -gencode=arch=compute_52,code=sm_52
+#        -gencode=arch=compute_60,code=sm_60
+#        -gencode=arch=compute_61,code=sm_61
+#        )
+#endif()
 
 
 set(CUDA_OPTS
